### PR TITLE
Desktop window

### PR DESCRIPTION
**Pull request by ocalvo**
Issue: #

## PR Type
What kind of change does this PR introduce?

- Feature

## What is the current behavior?

There is no support for WinUI only projects.
This change introduces DesktopWindow, and WinRT window with support for Xaml only content.

## What is the new behavior?
In managed or native projects can do the following:
```
        [STAThread]
        public static void Main()
        {
            using(var app = new App())
            {
                app.Run(new MainWindow());
            }
        }
```

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: 
- [X] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [ ] Contains **NO** breaking changes

### CODE REVIEW

File: Microsoft.Toolkit.Sample.DesktopWindow/Microsoft.Toolkit.Sample.DesktopWindow.csproj [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <Platforms>x86;x64</Platforms>
+    <!-- Workaround: https://github.com/microsoft/microsoft-ui-xaml/issues/798 -->
+    <TargetPlatformVersion>10.0.18362.0</TargetPlatformVersion>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
+  </PropertyGroup>
```
Comment (azchohfi) [inline Microsoft.Toolkit.Sample.DesktopWindow/Microsoft.Toolkit.Sample.DesktopWindow.csproj]: This is missing a `<IsPackable>false</IsPackable>`.
This project is generating a nupkg, and it shouldn't.

File: Microsoft.Toolkit.Win32.Xaml.targets [hunk 2]
```diff
@@ -0,0 +1,937 @@
...
+                $(AssemblyOriginatorKeyFile);
+                @(ReferencePath);
+                @(CompiledLicenseFile);
+                @(LinkResource);
+                @(EmbeddedDocumentation); 
+                $(Win32Resource);
+                $(Win32Manifest);
+                @(CustomAdditionalCompileInputs);
+                @(DesktopWindow);
+                @(DektopApp);
```
Comment (wjk) [inline Microsoft.Toolkit.Win32.Xaml.targets]: Typo here, should be `@(DesktopApp)`.

File: Microsoft.Toolkit.Win32.Xaml.targets [hunk 3]
```diff
@@ -0,0 +1,937 @@
...
+                $(ResolvedCodeAnalysisRuleSet)"
+                  
+     Outputs="@(DocFileItem);
+                 @(XamlIntermediateAssembly);
+                 @(_DebugSymbolsIntermediatePath);
+                 $(NonExistentFile);
+                 @(CustomAdditionalCompileOutputs)"
+     Condition="'@(DesktopWindow)' != '' Or '@(DektopApp)' != ''"
```
Comment (wjk) [inline Microsoft.Toolkit.Win32.Xaml.targets]: Ditto.

**Comment by marb2000** [comment]
Closing the PR given it was the inception of WinUI 3 in Desktop and this will never be part of the product.

**Pull request closed without merging**